### PR TITLE
chore: tsconfigs settings, strictness, comments, cleanup

### PR DESCRIPTION
**Pull request by NickKelly1**
- Extracts standard, applicable parts of tsconfigs from npm packages @tsconfig/node20 and @vue/tsconfig and removes the packages
- Organises and comments tsconfig files
- Makes tsconfigs slightly more strict

### Devop

* \[ ] Added entry to ./changelog
* \[ ] Add PR label

### CODE REVIEW

**Review by gamalielhere — APPROVED** [review]

File: nightwatch/tsconfig.json [hunk 1]
```diff
@@ -1,20 +1,49 @@
...
+
+    // Language & Environment
+    // TODO: Target the latest JS?
+    "target": "ESNext",               // https://www.typescriptlang.org/tsconfig/#target
+    "useDefineForClassFields": true,  // https://www.typescriptlang.org/tsconfig/#useDefineForClassFields
+    "lib": ["ESNext", "DOM"],         // https://www.typescriptlang.org/tsconfig/#lib
+
+    // Module
+    // TODO: does this need to be commonjs?
+    "module": "commonjs",           // https://www.typescriptlang.org/tsconfig/#module
```
Comment (olgakup) [inline nightwatch/tsconfig.json:27]: I am pretty sure we dont need this

File: nightwatch/tsconfig.json [hunk 2]
```diff
@@ -1,20 +1,49 @@
...
+    "useUnknownInCatchVariables": true,  // https://www.typescriptlang.org/tsconfig/#useUnknownInCatchVariables
+
+    // Interop constraints
+    "verbatimModuleSyntax": true,             // https://www.typescriptlang.org/tsconfig/#verbatimModuleSyntax
+    "esModuleInterop": true,                  // https://www.typescriptlang.org/tsconfig/#esModuleInterop
+    "forceConsistentCasingInFileNames": true, // https://www.typescriptlang.org/tsconfig/#forceConsistentCasingInFileNames
+
+    // Completeness
+    "skipLibCheck": true,  // https://www.typescriptlang.org/tsconfig/#skipLibCheck
+  },
```
Comment (olgakup) [inline nightwatch/tsconfig.json:48]: i would add allowSyntheticDefaultImports = true

File: package.json [hunk 3]
```diff
@@ -25,15 +25,13 @@
...
     "@nightwatch/vue": "^3.1.2",
-    "@tsconfig/node20": "^20.1.4",
     "@types/jsdom": "^21.1.7",
     "@types/node": "^20.16.11",
     "@vitejs/plugin-vue": "^5.1.4",
     "@vitest/eslint-plugin": "1.1.7",
     "@vue/eslint-config-prettier": "^10.0.0",
     "@vue/eslint-config-typescript": "^14.0.1",
     "@vue/test-utils": "^2.4.6",
-    "@vue/tsconfig": "^0.5.1",
```
Comment (olgakup) [inline package.json:36]: this guy has "resolveJsonModule": true, (we use json files for translations)

**Review by olgakup — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (NickKelly1) [reply package.json:36]: :+1:  I've added it to tsconfig.app.json

**Comment by NickKelly1** [comment]
Ready for re-review @olgakup

**Review by gamalielhere — APPROVED** [review]

**Pull request merged**